src/Sulu/Component/CustomUrl/Manager/CustomUrlManager.php

```diff
@@ -0,0 +1,158 @@
+{
+    /**
+     * @var DocumentManagerInterface
+     */
+    private $documentManager;
+
+    /**
+     * @var CustomUrlRepository
+     */
+    private $customUrlRepository;
+
+    /**
+     * @var PathBuilder
+     */
+    private $pathBuilder;
+
+    public function __construct(
+        DocumentManagerInterface $documentManager,
+        CustomUrlRepository $customUrlRepository,
+        PathBuilder $pathBuilder
+    ) {
+        $this->documentManager = $documentManager;
+        $this->customUrlRepository = $customUrlRepository;
+        $this->pathBuilder = $pathBuilder;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create($webspaceKey, array $data, $locale = null)
+    {
+        $document = $this->documentManager->create('custom_urls');
+        $this->bind($document, $data, $locale);
+
+        $this->documentManager->persist(
+            $document,
+            $locale,
+            [
+                'parent_path' => $this->getItemsPath($webspaceKey),
+                'node_name' => Urlizer::urlize($document->getTitle()),
+                'load_ghost_content' => true,
+            ]
+        );
+
+        return $document;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function readList($webspaceKey)
+    {
+        // TODO pagination
+
+        return $this->customUrlRepository->findList($this->getItemsPath($webspaceKey));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function read($uuid, $locale = null)
+    {
+        return $this->documentManager->find($uuid, $locale, ['load_ghost_content' => true]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function update($uuid, array $data, $locale = null)
+    {
+        $document = $this->read($uuid, $locale);
+        $this->bind($document, $data, $locale);
+
+        $this->documentManager->persist($document, $locale);
+
+        return $document;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function delete($uuid)
+    {
+        $this->documentManager->remove($this->read($uuid));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFields()
+    {
+        return [
+            'title' => ['property' => 'title'],
+            'published' => ['property' => 'published'],
+            'baseDomain' => ['property' => 'baseDomain'],
+            'domainParts' => ['property' => 'domainParts', 'type' => 'json_array'],
+            'target' => ['property' => 'target', 'type' => 'reference'],
+            'multilingual' => ['property' => 'multilingual'],
+            'canonical' => ['property' => 'canonical'],
+            'redirect' => ['property' => 'redirect'],
+            'targetLocale' => ['property' => 'targetLocale'],
+        ];
+    }
+
+    /**
+     * Bind data array to given document.
+     *
+     * @param CustomUrlDocument $document
+     * @param array $data
+     * @param string $locale
+     */
+    private function bind(CustomUrlDocument $document, $data, $locale)
```

Comment:
But it does this job. And it should do it in two lines of code. We may
well replace it, but it is there now.

On Fri, Jan 22, 2016 at 02:40:50AM -0800, Wachter Johannes wrote:
>    In [1]src/Sulu/Component/CustomUrl/Manager/CustomUrlManager.php:
> 
>  > +            'target' => ['property' => 'target', 'type' => 'reference'],
>  > +            'multilingual' => ['property' => 'multilingual'],
>  > +            'canonical' => ['property' => 'canonical'],
>  > +            'redirect' => ['property' => 'redirect'],
>  > +            'targetLocale' => ['property' => 'targetLocale'],
>  > +        ];
>  > +    }
>  > +
>  > +    /**
>  > +     * Bind data array to given document.
>  > +     *
>  > +     * @param CustomUrlDocument $document
>  > +     * @param array $data
>  > +     * @param string $locale
>  > +     */
>  > +    private function bind(CustomUrlDocument $document, $data, $locale)
> 
>    but this FormFactory is the form-component which we discuss the whole time
>    and which we are not sure (i think we are mostly sure) that it is not the
>    best solution for our needs and therefor we should not use in my opinion
>    /cc [2]@danrot
> 
>    —
>    Reply to this email directly or [3]view it on GitHub.
> 
>    Reverse link: [4]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/sulu-io/sulu/pull/1927#discussion_r50523347
>    2. https://github.com/danrot
>    3. https://github.com/sulu-io/sulu/pull/1927/files#r50523347
>    4. https://github.com/sulu-io/sulu/pull/1927/files#r50523347